### PR TITLE
Route disk usage requests to promotable shards

### PR DESCRIPTION
**Pull request by dnhatn**
If the flush parameter of a disk usage request is true, the target shards will flush before analyzing the disk usage. Therefore, it is necessary to route a request with flush=true to the promotable copies.

Alternative approach is to perform a disk usage request in two steps: (1) flush the target indices if flush=true, (2) analyze disk usage on the target indices after flushed. This approach would support any shard roles.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/TransportAnalyzeIndexDiskUsageAction.java [hunk 1]
```diff
@@ -216,8 +217,9 @@ protected GroupShardsIterator<ShardIterator> shards(
         AnalyzeIndexDiskUsageRequest request,
         String[] concreteIndices
     ) {
+        final Predicate<ShardRouting> shardRole = request.flush ? ShardRouting::isPromotableToPrimary : ShardRouting.ANY_ROLE;
         final GroupShardsIterator<ShardIterator> groups = clusterService.operationRouting()
-            .searchShards(clusterState, concreteIndices, null, null);
+            .searchShards(clusterState, concreteIndices, shardRole, null, null);
```
Comment (dnhatn) [inline server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/TransportAnalyzeIndexDiskUsageAction.java:222]: This is the main change of this PR.

File: server/src/main/java/org/elasticsearch/action/termvectors/TransportTermVectorsAction.java [hunk 2]
```diff
@@ -63,7 +64,7 @@ protected ShardIterator shards(ClusterState state, InternalRequest request) {
         if (request.request().doc() != null && request.request().routing() == null) {
             // artificial document without routing specified, ignore its "id" and use either random shard or according to preference
             GroupShardsIterator<ShardIterator> groupShardsIter = clusterService.operationRouting()
-                .searchShards(state, new String[] { request.concreteIndex() }, null, request.request().preference());
+                .searchShards(state, new String[] { request.concreteIndex() }, ShardRouting.ANY_ROLE, null, request.request().preference());
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/action/termvectors/TransportTermVectorsAction.java:67]: question: why does this use ANY_ROLE as opposed to searchable?

**Review by javanna — COMMENTED** [review]
left a small question, also do we need additional tests for this change, to verify that the role predicate is applied, or do we have tests already?

**Comment by dnhatn** [comment]
@elasticmachine update branch

**Pull request closed without merging**